helpers/stable_swap_math/mod.rs

```diff
@@ -88,9 +93,16 @@ fn compute_d_next(
     n_add_one: u32,
 ) -> Result<U256, MathError> {
     let mut d_prod = d_prev;
-    // d_prod = ... * [d_prev / (x_(i) * n)] * ...
-    // where i in (0,n)
+    // [nit]
+    // I'd prefer having a comment this way: d_prod = d_prev^(n+1) / (n^n * Prod{amounts_i})
     for amount in amounts {
+        // [audit]: I'm still not sure how to best multiply and divide by n^n
+        // In curve, they've changed this 6 months ago to dividing by n^n at the end:
+        // https://github.com/curvefi/stableswap-ng/commit/7ac164bb7d9cf800d50adf549f10539684db69b8
+        // (supposedly after some large audit).
+        // I think it makes sense, as for n <= 8, the value of n^n is not too big, so we don't save much
+        // by dividing product by it when iterating (as it is currently), but we might
+        // run into some precision issues instead, when amount*n ~ d_prod.
```

Comment:
<3

Reply:
argument in favor of dividing by `n^n` is that (simplest example for 2 tokens) when you have `A, B` and want to compute `AB/n^n = AB/4`, it's possible that `A` is divisible by 4 and `B` is not divisible by 2. Then you don't lose any precision when dividing `AB` by 4, but you lose when you divide `B` by 2.